### PR TITLE
Add Qwen3-VL-2B E2E test scripts

### PR DESCRIPTION
**Pull request by chiajunglien**
# Description

This PR introduces end-to-end (E2E) testing and validation pipelines for the Qwen3-VL-2B model in MaxText and adjusts the config of Qwen3-30b E2E script.

**Note on `scan_layers=false`**:
Qwen3-VL-2B requires `scan_layers=false` due to its Deepstack visual embedding injection design. Because the visual tokens are interwoven deeply across specific layers, the transformer blocks are not computationally identical and therefore cannot be wrapped inside a homogeneous `jax.lax.scan` loop.

# Tests (Testing on v6e-8 TPU VM)

Multimodal SFT: 
```bash
export RUN_ID=$(date +%Y-%m-%d-%H-%M)
bash tests/end_to_end/tpu/qwen3/vl_2b/test_qwen3_to_mt.sh $RUN_ID
bash tests/end_to_end/tpu/qwen3/vl_2b/test_qwen3_multimodal_sft.sh $RUN_ID
```

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

**Comment by YixuanWang-99** [comment]
can you squash your commits?

**Comment by chiajunglien** [comment]
> can you squash your commits?

Thanks for the reminder! I've squashed the commits into one and updated the PR.

File: tests/end_to_end/tpu/qwen3/vl_2b/test_qwen3_rl.sh
Comment (SurbhiJainUSC) [inline tests/end_to_end/tpu/qwen3/vl_2b/test_qwen3_rl.sh:1]: Qwen3-vl-2b is a vision-language model. I don't think we support multimodality with RL. LEt's just not add text-only tests for this model.

File: tests/end_to_end/tpu/qwen3/vl_2b/test_qwen3.sh
Comment (SurbhiJainUSC) [inline tests/end_to_end/tpu/qwen3/vl_2b/test_qwen3.sh:1]: @hengtaoguo - can you also review this PR?

**Review by YixuanWang-99 — APPROVED** [review]

File: tests/end_to_end/tpu/qwen3/vl_2b/test_qwen3_rl.sh
Comment (chiajunglien) [reply tests/end_to_end/tpu/qwen3/vl_2b/test_qwen3_rl.sh:1]: Got it. I've removed `pretrain`, `SFT`, and `RL` tests and kept `multimodal_sft`, as those three currently only support text-only.

**Review by SurbhiJainUSC — APPROVED** [review]

File: tests/end_to_end/tpu/qwen3/vl_2b/test_qwen3_multimodal_sft.sh [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    max_prefill_predict_length=1024 \
+    max_target_length=2048 \
+    steps=5 \
+    scan_layers=false \
+    async_checkpointing=False \
+    float32_qk_product=True \
+    float32_logits=True \
+    dataset_type=hf \
+    hf_path=parquet \
+    hf_train_files=gs://aireenmei-multipod/dataset/hf/chartqa/train-* \
```
Comment (hengtaoguo) [inline tests/end_to_end/tpu/qwen3/vl_2b/test_qwen3_multimodal_sft.sh]: Could you make a copy of this folder into gs://maxtext-dataset bucket like this [example](https://github.com/AI-Hypercomputer/maxtext/blob/42045bbb213900b737144e26641d6c022baf1438/tests/end_to_end/tpu/llama3/8b/2_test_llama3_8b.sh#L34)? Thanks!

**Review by hengtaoguo — APPROVED** [review]
Thanks for driving the tests! 

Could you also update the PR description since multimodal doesn't have RL support right now? Thanks!

File: tests/end_to_end/tpu/qwen3/vl_2b/test_qwen3_multimodal_sft.sh (hunk 1 above)
Comment (chiajunglien) [reply tests/end_to_end/tpu/qwen3/vl_2b/test_qwen3_multimodal_sft.sh]: I've removed the RL mention from the PR description. I've also updated the dataset path to `gs://maxtext-dataset/hf/chartqa`.
